src/core/shader.js

```diff
@@ -61,18 +62,19 @@ Shader.prototype = {
   },
 
   initVariables: function (data, type) {
-    var self = this;
     var variables = {};
     var schema = this.schema;
     var schemaKeys = Object.keys(schema);
     schemaKeys.forEach(processSchema);
     function processSchema (key) {
       if (schema[key].is !== type) { return; }
       var varType = propertyToThreeMapping[schema[key].type];
-      var varValue = schema[key].parse(data[key] || schema[key].default);
-      variables[key] = {
+      var varKey = key;
+      // For textures, by convention data.src becomes uniform sampler2D map.
+      if (varKey === 'src' && type === 'uniform' && varType === 't') { varKey = 'map'; }
+      variables[varKey] = {
         type: varType,
-        value: self.parseValue(schema[key].type, varValue)
+        value: undefined // let updateVariables handle setting these
```

Comment:
not sure how to take this over to #2199... 

that is why the comment is there.  `update` will be triggered, which calls `updateVariables`, which will only take action if the value is changed... so it is important that the value from `init` not be the correct value, else no action will be taken by `updateVariables`